### ISSUE TITLE
Show type class or equivalent?

### ISSUE THREAD

**Issue by johnhungerford** [getkyo/kyo#907]
Given the extensive use of opaque types, I think it might make sense to provide some ad-hoc polymorphic utility for generating string representations of objects rather than using `toString`.

Some issues I've run into:
- Logging a `Result` and seeing a `()` or logging a `Maybe` and seeing a `23` instead of a `Present(23)` is confusing on its face when we're used to seeing the wrappers
- It also makes it hard to distinguish logs when in one place we're logging a result and in another we're logging an extracted value. 
- We can't see the level of nesting of the value for `Result` and `Maybe`
- Logging an effect and seeing a pure value can make it look like an effect has been fully handled when it hasn't

The obvious solution is a `Show` type class, with derived instances for your usual types and explicit given instances for kyo's types:
```scala
trait Show[A]:
  inline def show(value: A): String

object Show:
   // This could be replaced with ADT derivation if we want to provide some custom
   // view for case classes
   given default[A]: Show[A] with { inline def show(value: A): String = value.toString }
   
...

object Maybe:
  given [A](using ash: Show[A]): Show[Maybe[A]] with
    inline def show(value: Maybe[A]): String =
      value.fold("Absent")(a => s"Present(${ash.show(a)})")
```

**Comment by DamianReeves** [comment]
I wonder if a Rust style distinction of `Display` and `Debug` would make sense here. Either way it would be great if it was a `zio-prelude`/Rust style debug that had some formatting help. 

Agree with the overall ask either way though.

**Comment by johnhungerford** [comment]
@DamianReeves we just merged a simpler version that can render kyo types to `Text`. We had some chat in the PR about adding some more logic -- maybe supporting multiple modes of text rendering -- but that will be down the line.

**Comment by johnhungerford** [comment]
Closed with #914
